lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1449,6 +1453,28 @@ _assignScriptRedeemers pparams ti resolveInput redeemers tx =
                 }
             }
 
+eraFromPParams :: ProtocolParameters -> AnyCardanoEra
+eraFromPParams protocolParams =
```

Comment:
Apologies for the missing documentation in `EraInfo`: I pushed #3366.

Telling the current era from `eraInfo` is impractical because it may contain `Just futureEpoch` for eras with hard-forks which are confirmed but hasn't yet occurred.

But I believe what you have here also doesn't work because we still know when past eras started when a new era start: you probably hit the top `Just _byron` case even if we're in Babbage.

Reply:
If you need the current node era: perhaps it would be an idea to add an `era` field to `W.ProtocolParameters`? @jonathanknowles _might_ be interested in this as well.

Although I suspect you're correct with:
> Pretty sure we'll have to use the transaction era, not the current node era.

I believe they should match with the old tx workflow, but not necessarily the new one (in particular we can call balanceTx with an old era tx)

Reply:
Where `balanceTx` calls coin-selection / TransactionLayer functions `era` is available in the type level, and can be brought down with `cardanoEra @era`.

For the old tx workflow I hope it shouldn't be too cumbersome to pipe through the actual `era` used to construct the tx. (related comment https://github.com/input-output-hk/cardano-wallet/pull/3362#discussion_r909479868)

Reply:
Ok, I'm not 100% sure I've got it right, but I've tried something new:

- Where I have the era of the Tx available, use it.
- Where we are constructing a Tx, just pull from current node era.

Reply:
`selectAssets` now takes era as a parameter.

Reply:
As for the old transaction workflow... I'm not sure... I don't think the transaction era is available? Because we're creating it.